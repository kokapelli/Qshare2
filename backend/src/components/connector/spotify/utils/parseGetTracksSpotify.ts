import { QShareTrack, QShareArtist } from "../../../music/musicTypes";

interface SpotifyTrack {
    name: string;
    uri: string;
    artists: SpotifyArtist[];
    album: SpotifyAlbum;
    popularity: number;
}

interface SpotifyArtist {
    uri: string;
    name: string;
}

interface SpotifyAlbum {
    images: SpotifyAlbumImage[];
}

interface SpotifyAlbumImage {
    width: number;
    height: number;
    url: string;
}

interface SpotifyGetTracksResponse {
    tracks: {
        name: string
        items: SpotifyTrack[]
        limit: number
    };
}

export interface QShareGetTrackResponse extends QShareTrack {
    popularity: number;
}

export interface QShareGetTracksResponse {
    limit: number;
    tracks: QShareGetTrackResponse[];
}

export const parseGetTracksSpotify = (data: SpotifyGetTracksResponse) => {
    const response: QShareGetTracksResponse = {} as any;
    const tracks: QShareGetTrackResponse[] = [];

    response.tracks = tracks;
    response.limit = data.tracks.limit;
    for (const track of Array.from(data.tracks.items)) {
        const newTrack: QShareGetTrackResponse = {} as any;
        newTrack.name = track.name;

        const newArtists: QShareArtist[] = [];
        for (const artist of track.artists) {
            const newArtist: SpotifyArtist = {
                uri: artist.uri,
                name: artist.name
            };

            newArtists.push(newArtist);
        }

        newTrack.artists = newArtists;

        const newImages = [];
        for (const image of track.album.images) {
            const newImage: SpotifyAlbumImage = {
                height: image.height,
                width: image.width,
                url: image.url
            };
            newImages.push(newImage);
        }

        newTrack.images = newImages;

        newTrack.uri = track.uri;
        newTrack.popularity = track.popularity;
        tracks.push(newTrack);
    }

    return response;
};
