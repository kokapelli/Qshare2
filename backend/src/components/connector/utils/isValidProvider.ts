import { SPOTIFY_PROVIDER } from "../../constants/providers";

const isValidProvider = (provider: string) => {
    return provider === SPOTIFY_PROVIDER;
};

export { isValidProvider };