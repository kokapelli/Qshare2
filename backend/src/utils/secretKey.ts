import crypto from "crypto";

const SECRET_KEY = crypto.randomBytes(256);

export default SECRET_KEY;
