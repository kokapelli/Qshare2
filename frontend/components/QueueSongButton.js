import React, { Component } from 'react';
import { Text, StyleSheet, View, } from 'react-native';
import _ from 'lodash';

export class QueueSongButton extends Component{
    render(){
      const {track} = this.props;
      const {artists} = track;
      const name = _.truncate(track.name, {length:20});
      const artist = artists[0]? _.truncate(artists[0].name, {length:20}):'Undefined';
        return(
            <View style={styles.queueItem}>
                <View style={styles.songName}>
                    <Text testID='songDetails' style={styles.songText}>
                        {name} - {artist}
                    </Text>
                </View>
          </View>
        );
    }
}

const styles = StyleSheet.create({
    queueItem: {
        height: 60,
        borderWidth: 1,
        paddingLeft: 30,
        paddingRight: 30,
        borderRadius: 50,
        marginTop: 5,
        borderColor: '#383838',
        flexDirection: 'row',
        justifyContent: 'center',
        backgroundColor: '#282828',
        margin: 3,
      },
    songName: {
        flex: 4,
        justifyContent: 'center',
        backgroundColor: '#282828',
    },
    songText: {
        color: 'white',
        fontSize: 20,
    },
    voteButton: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#ffea00',
        borderTopLeftRadius: 20,
        borderBottomLeftRadius: 20,
    },
  });

export default QueueSongButton