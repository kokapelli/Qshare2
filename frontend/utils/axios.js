import Axios from "axios"
import { settings } from "../settings/settings";

const axios = Axios.create()
axios.defaults.headers.common['Provider'] = settings.provider

export default axios