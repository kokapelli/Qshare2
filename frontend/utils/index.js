export const findByTestAttr = (component, attr)  => {
    const wrapper = component.find(`[testID='${attr}']`);
    return wrapper
  }
  