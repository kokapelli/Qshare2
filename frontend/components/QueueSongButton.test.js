import { QueueSongButton } from './QueueSongButton'
import { shallow, configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { findByTestAttr } from '../utils/'

configure({ adapter: new Adapter() });

const track = {
    name: "Qshare",
    artists: [
        {
            name: "Hampus"
        },
        {
            name: "Brent"
        },
        {
            name: "Valter"
        }
    ]
}

describe('QueueSongButton Snapshot', () => {

    it('matches the snapshot', () => {
        const tree = renderer.create(
        <QueueSongButton track={track}/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('QueueSongButton Units', () => {

    beforeEach(()=>{
       this.component = shallow(<QueueSongButton track={track}/>)
    })

    it('component renders properly', () => {
        const rendered = renderer.create(<QueueSongButton track={track}/>).toJSON()
        expect(rendered).toBeTruthy()
    });

    it('Song Details should be correct given mock input', () => {
        const details = findByTestAttr(this.component, 'songDetails')
        const expected = [track.name, " - ", track.artists[0].name]
        expect(details.props().children).toEqual(expected)
        })

})