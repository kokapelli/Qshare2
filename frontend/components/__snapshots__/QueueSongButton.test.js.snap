// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueueSongButton Snapshot matches the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#282828",
      "borderColor": "#383838",
      "borderRadius": 50,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "margin": 3,
      "marginTop": 5,
      "paddingLeft": 30,
      "paddingRight": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#282828",
        "flex": 4,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
      testID="songDetails"
    >
      Qshare
       - 
      Hampus
    </Text>
  </View>
</View>
`;
